invenio_oauthclient/views/client.py

```diff
@@ -64,6 +65,28 @@ def post_ext_init(state):
                        'invenio_oauthclient/settings/base.html'))
 
 
+@blueprint.route("/login")
+def auto_redirect_login(*args, **kwargs):
+    """Handle automatic redirect to external auth service if configured."""
+    local_login_enabled = current_app.config.get(
+        "ACCOUNTS_LOCAL_LOGIN_ENABLED", False
+    )
+    auto_redirect_enabled = current_app.config.get(
+        "OAUTHCLIENT_AUTO_REDIRECT_TO_EXTERNAL_LOGIN", False
+    )
+    would_redirect = auto_redirect_enabled and not local_login_enabled
+    remote_apps = list(current_oauthclient.oauth.remote_apps)
+
+    if would_redirect and len(remote_apps) == 1:
+        # if local login is disabled and we only have one OAuth2 remote app
+        # configured, we forward directly to that
+        url = url_for("invenio_oauthclient.login", remote_app=remote_apps[0])
+        return redirect(url)
+
+    else:
+        return base_login(*args, **kwargs)
```

Comment:
There might be an issue. Imagine the scenario:
- I have configured for `invenio-accounts` `ACCOUNTS_LOGIN_VIEW_FUNCTION = myloginfunc`
- I install `invenio-oauthclient`, and it overrides it to `ACCOUNTS_LOGIN_VIEW_FUNCTION = auto_redirect_login`
- if I have `ACCOUNTS_LOCAL_LOGIN_ENABLED = True`, it will return `base_login` and not `myloginfunc`

or maybe this is not an issue because in this case I will have to redefine `auto_redirect_login` to return`myloginfunc` ...

Reply:
hmm... wouldn't the configured value (via `invenio.cfg`, i suppose?) then still take precedence over the default values provided by any of the modules?